### PR TITLE
fix: 🐛 Added transact write to dynamoClient

### PR DESCRIPTION
**Pull request by priyanshu966106**
## BACKEND-STORY_ID One-sentence summary of changes
Transact Write was missing from dynamodb client 

#### Is it a breaking change?: NO / YES
NO

### Why did you make these changes?
Added transact write 

### What's changed in these changes?

Added a new function and used it in dynamodb Client 

### What do you especially want to get reviewed?

transact_write.ts

### Submission Type

* [ ] Bugfix
* [X] New Feature
* [ ] Refactor

### All Submissions

* [X ] Have you added an explanation of what your changes?
* [ ] Have you written new tests for your changes, as applicable?
* [X ] Have you checked potential side effects that could make bad impacts to other services?

### New Features

* [ ] Have you configured CI/CD properly?
* [ ] Have you configured optimal memory size and timeouts of Lambda Function?
* [ ] Have you grant required permissions (e.g. IAM Policy, VPC Security Group)?
* [ ] Have you made required resources (e.g. DynamoDB Table, RDS Cluster)?
* [ ] Does it requires native addons dependency?

### CODE REVIEW

File: src/query/transact_write.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                }
+            ).promise()
+          return res
+        }),
+    )
+
+    return results
+  } catch (e) {
+    // tslint:disable-next-line
+    console.log(`Dynamo-Types TransactWrite - ${JSON.stringify(requests, null, 2)}`)
```
Comment (visheshd) [inline src/query/transact_write.ts:26]: Please use the same pattern used elsewhere
```typescript

const log = input.log || console['log']

```

File: src/document-client.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { DynamoDB } from 'aws-sdk'
 import * as _ from 'lodash'
 import { batchWrite } from './query/batch_write'
+import {transactWrite} from './query/transact_write'
```
Comment (visheshd) [inline src/document-client.ts:4]: Formatting around braces

File: src/query/transact_write.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { DynamoDB } from 'aws-sdk'
+import * as _ from 'lodash'
+
+
+export async function transactWrite(
+  documentClient: DynamoDB,
+  requests: DynamoDB.TransactWriteItem[],
+) {
+  try {
+    const results = await Promise.all(
```
Comment (visheshd) [inline src/query/transact_write.ts:10]: Also, chunk similar to batchWrite

Comment (priyanshu966106) [reply src/query/transact_write.ts:12]: Sure added chunking

File: src/document-client.ts (hunk 2 above)
Comment (priyanshu966106) [reply src/document-client.ts]: Sure thanks corrected

**Comment by benhutchins** [comment]
First off, thank you for submitting this PR @priyanshu966106 and reviewing it @visheshd. I'm just going to do a bit of testing before I merge.

**Review by benhutchins — COMMENTED** [review]
This appears to work as expected, but I spent some time thinking about the implementation of the batch write and this new transact write. Part of the usefulness of these within DynamoDB is that you can perform a batch write of several documents from different tables, which can be useful if you are creating a several types of records at once.

This new transact write method makes usefulness of being able to write across several tables at once far more useful and it would be nice to expose that. An operation could touch several tables; and you want the entire operation to work successfully or fail. We can expose that with transactWrite where the process is atomic, if any part fails the entire operation will have no affect. This is a very useful feature, but it is really more useful if it was exposed in a way that worked across several tables.

So I worked on an updated proposal for this, which works using a class that allows you to build I'm thinking it might be useful to expose new classes globally, like:

```typescript
const transaction = new Dyngoose.Transaction()
transaction.save(item) // intelligently will use Put or Update similar to the Table.save() method
transaction.put(...items) // you could use .put or .update explicitly to be more discrete
transaction.delete(item) // would 
transaction.conditionCheck(item, conditions) // adds a condition to an item not being updated
await transaction.commit()
```

Similarly, for batch write operations, you could use:

```typescript
const batch = new Dyngoose.BatchWrite()
batch.put(item)
batch.delete(item)
await batch.commit()
```

It might be useful to do something similar with the BatchGet operation, which also supports reading from several tables at once; but I think the use case is a bit more limited and the implementation would be a bit difficult to deal with since the response would contain a mix and the order isn't guaranteed. As it stands, the current `batchGet` operation does not handle the `UnprocessedKeys` correctly. So that will come later.

**Comment by visheshd** [comment]
This is cool

**Pull request merged**